File: datasets/lua/HumanEval_66_digitSum.lua
-- Task
-- Write a function that takes a string as input and returns the sum of the upper characters only'
-- ASCII codes.
-- Examples:
-- digitSum("") => 0
-- digitSum("abAB") => 131
-- digitSum("abcCd") => 67
-- digitSum("helloE") => 69
-- digitSum("woArBld") => 131
-- digitSum("aAaaaXa") => 153
local function digitSum(s)
  local sum = 0
  for i = 1, #s do
    local c = s:sub(i, i)
    if c:match("%u") then
      sum = sum + c:byte()
    end
  end
  return sum
end
-- Unit tests below

lu = require('luaunit')

function test_humaneval()
local candidate = digitSum
    lu.assertEquals(candidate(''), 0)
    lu.assertEquals(candidate('abAB'), 131)
    lu.assertEquals(candidate('abcCd'), 67)
    lu.assertEquals(candidate('helloE'), 69)
    lu.assertEquals(candidate('woArBld'), 131)
    lu.assertEquals(candidate('aAaaaXa'), 153)
    lu.assertEquals(candidate(' How are yOu?'), 151)
    lu.assertEquals(candidate('You arE Very Smart'), 327)
end

os.exit(lu.LuaUnit.run())

File: datasets/rb/HumanEval_66_digitSum.rb
# Task
# Write a function that takes a string as input and returns the sum of the upper characters only'
# ASCII codes.
# Examples:
# digitSum("") => 0
# digitSum("abAB") => 131
# digitSum("abcCd") => 67
# digitSum("helloE") => 69
# digitSum("woArBld") => 131
# digitSum("aAaaaXa") => 153
def digitSum(s)
  s.scan(/[A-Z]/).map(&:ord).reduce(:+)
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_digitSum
    candidate = method(:digitSum)
    assert_equal(0, candidate.call(''))
    assert_equal(131, candidate.call('abAB'))
    assert_equal(67, candidate.call('abcCd'))
    assert_equal(69, candidate.call('helloE'))
    assert_equal(131, candidate.call('woArBld'))
    assert_equal(153, candidate.call('aAaaaXa'))
    assert_equal(151, candidate.call(' How are yOu?'))
    assert_equal(327, candidate.call('You arE Very Smart'))
  end
end
